src/transformers/modeling_transfo_xl.py

```diff
@@ -827,6 +827,7 @@ def tie_weights(self):
         # sampled softmax
         if self.sample_softmax > 0:
             if self.config.tie_weight:
+                # TODO: this does not make sense -> transformer.word_emb.weight does not exist
```

Comment:
@LysandreJik @thomwolf . I didn't look deep enough into the paper and the code to understand everything fully, but even though all tests pass - there is a problem here. `self.out_layer.weight` is of dim `[267735, 1024]` where as `transformer.word_emd` is a list of differently shaped Embeddings -> not easy to copy weights here.

Reply:
I see.
We can remove the `sample_softmax` option if it makes the code cleaner.
The Transformer-XL codebase is definitely the most "researchy" one in the library and we've already had a few passes to remove dead code in it. Now could be a good time to remove the dead code related to Sampling Softmax (it's not used in pretrained models).